example/heatEquation2D/src/heatEquation2D.cpp

```diff
@@ -0,0 +1,198 @@
+    }
+    // Set Dim and Idx type
+    using Dim = alpaka::DimInt<2u>;
+    using Idx = uint32_t;
+
+    // Define the accelerator
+    using Acc = alpaka::TagToAcc<TAccTag, Dim, Idx>;
+    std::cout << "Using alpaka accelerator: " << alpaka::getAccName<Acc>() << std::endl;
+
+    // Select specific devices
+    auto const platformHost = alpaka::PlatformCpu{};
+    auto const devHost = alpaka::getDevByIdx(platformHost, 0);
+    auto const platformAcc = alpaka::Platform<Acc>{};
+    auto const devAcc = alpaka::getDevByIdx(platformAcc, 0);
+
+    // simulation defines
+    // Parameters
+    constexpr alpaka::Vec<Dim, Idx> numNodes{64, 64}; // {Y, X}
+    constexpr alpaka::Vec<Dim, Idx> haloSize{2, 2};
+    constexpr alpaka::Vec<Dim, Idx> extent = numNodes + haloSize;
+
+    constexpr uint32_t numTimeSteps = 100;
+    constexpr double tMax = 0.001;
+    // x, y in [0, 1], t in [0, tMax]
+    constexpr double dx = 1.0 / static_cast<double>(extent[1] - 1);
+    constexpr double dy = 1.0 / static_cast<double>(extent[0] - 1);
+    constexpr double dt = tMax / static_cast<double>(numTimeSteps);
+
+    // Check the stability condition
+    constexpr double r = dt / std ::min(dx * dx, dy * dy);
+    if constexpr(r > 0.5)
+    {
+        std::cerr << "Stability condition check failed: dt/min(dx^2,dy^2) = " << r
+                  << ", it is required to be <= 0.5\n";
+        return EXIT_FAILURE;
+    }
+
+    // Initialize host-buffer
+    // This buffer will hold the current values (used for the next step)
+    auto uBufHost = alpaka::allocBuf<double, Idx>(devHost, extent);
+
+    // Accelerator buffer
+    auto uCurrBufAcc = alpaka::allocBuf<double, Idx>(devAcc, extent);
+    auto uNextBufAcc = alpaka::allocBuf<double, Idx>(devAcc, extent);
+
+    // Set buffer to initial conditions
+    initalizeBuffer(uBufHost, dx, dy);
+
+    // Select queue
+    using QueueProperty = alpaka::NonBlocking;
+    using QueueAcc = alpaka::Queue<Acc, QueueProperty>;
+    QueueAcc queue1{devAcc};
+    QueueAcc queue2{devAcc};
+
+    // Copy host -> device
+    alpaka::memcpy(queue1, uCurrBufAcc, uBufHost);
+    alpaka::wait(queue1);
+
+    // Define a workdiv for the given problem
+    constexpr alpaka::Vec<Dim, Idx> elemPerThread{1, 1};
+    // Appropriate chunk size to split your problem for your Acc
+    constexpr alpaka::Vec<Dim, Idx> chunkSize{16u, 16u};
+    constexpr alpaka::Vec<Dim, Idx> chunkSizeWithHalo{chunkSize[0] + haloSize[0], chunkSize[1] + haloSize[1]};
+
+    auto const maxThreadsPerBlock = alpaka::getAccDevProps<Acc>(devAcc).m_blockThreadExtentMax;
+    auto const threadsPerBlock = maxThreadsPerBlock.prod() < chunkSize.prod() ? maxThreadsPerBlock : chunkSize;
+
+    constexpr alpaka::Vec<Dim, Idx> numChunks{
+        (extent[0] - haloSize[0] - 1) / chunkSize[0] + 1,
+        (extent[1] - haloSize[1] - 1) / chunkSize[1] + 1};
+
+    static_assert(
+        (extent[0] - haloSize[0]) % chunkSize[0] == 0 && (extent[1] - haloSize[1]) % chunkSize[1] == 0,
```

Comment:
checking `numNodes/chunkSize` would be enough.

Reply:
changed